src/west/commands.py

```diff
@@ -351,7 +351,7 @@ def _commands_module_from_file(file):
 
 # Cache which maps files implementing extension commands to their
 # imported modules.
-_EXT_MODULES_CACHE = {}
+_EXT_MODULES_CACHE = {} # type: ignore
```

Comment:
How is this used by this PR? Just curious why it needs an annotation but, say _EXT_MODULES_NAME_IT does not.

Reply:
This is again `mypy` complaining about missing type annotation. The reason why it complains for this one and not others... I don't know. I guess it is related somehow with type inference performed by mypy.

```
src/west/commands.py:354: error: Need type annotation for '_EXT_MODULES_CACHE' (hint: "_EXT_MODULES_CACHE: Dict[<type>, <type>] = ...")
```

Anyway, I added the types.

Reply:
@mbolivar-nordic I forgot that this error comes up when annotating this one:
```
SyntaxError: annotated name '_EXT_MODULES_CACHE' can't be global
```
I'll investigate.

Reply:
Seems like forward declaring globals is a workaround, so I moved globals to the top...